woocommerce/woocommerce-ios
# Product description AI announcement modal: show the modal

Part of #10021 


## Why


We want to increase the visibility of the product description AI feature, and are showing a modal that has the same design as JITM NMncL2ViLjomfLhuyEhVM1-fi-236_84398. Originally, we wanted to reuse one of the two existing modals - JITM and What's New announcement modal. But each of the remote modal systems has limitations, and we decided to implement a modal locally pecCkj-Cr-p2#comment-382 after syncing JITM and no modal is being shown.

## How

It's not ideal to have a third kind of modal on the dashboard as the logic gets quite complicated. I mostly followed how JITM modal was implemented with some modifications as the changes are already large (sorry for going over 500 diffs 🙇🏻‍♀️). This PR just includes the changes for showing the modal, the CTA tap event and analytics will be in a future PR.

🗒️  There's a known issue with the existing JITM overlay. The modal is not scrollable, which means the modal can get cut off with a larger font or in landscape mode. I tried `scrollVerticallyIfNeeded` but I think something in the pre-existing `ModalOverlay` is preventing it from working. I'll look into this improvement in a future subtask.

🗒️ I thought about reusing the JITM UI `JustInTimeMessageModal`, but decided not to touch the JITM code since it's nontrivial to test it.

- The modal view `LocalAnnouncementModal` was created with its view model `LocalAnnouncementViewModel`
- `LocalAnnouncementsProvider` was created to load a local announcement
- `DashboardViewModel` and `DashboardViewController` were changed to load the local announcement after syncing JITM

## Testing instructions


- Launch the app in the logged-in state and connected to a WPCOM store --> a modal should be shown about the product description AI after seeing `📣 Announcements synced, but nothing received.`/` 📣 There are no announcements to show!` logs in the console
- Tap outside of the modal to make it disappear
- Switch to a self-hosted store --> no modal should be shown
- Switch back to a WPCOM store --> a modal should be shown shortly
- Relaunch the app --> a modal should be shown shortly
- Tap on either CTA in the modal --> the modal should be dismissed
- Relaunch the app --> no modal should be shown after it was dismissed

## Screenshots


dark | light
-- | --
![Simulator Screenshot - iPhone 14 - 2023-06-28 at 21 35 30](https://github.com/woocommerce/woocommerce-ios/assets/1945542/9bdfdc6b-4b5d-4a35-b13b-e9b86c83b436) | ![Simulator Screenshot - iPhone 14 - 2023-06-28 at 21 35 40](https://github.com/woocommerce/woocommerce-ios/assets/1945542/b730e66d-3dd7-4a4a-9945-42f781b48705)


---
